regions/core/compound.py

```diff
@@ -29,16 +29,21 @@ class CompoundSkyRegion(SkyRegion):
     Represents the logical combination of two regions in sky coordinates.
     """
 
-    def __init__(self, region1, operator, region2):
+    def __init__(self, region1, region2, operator):
```

Comment:
The reason I had operator in the center is because it makes it clearer how to deal with non-symmetric cases (e.g. region 1 without region 2)